src/snowflake/snowpark/functions.py

```diff
@@ -8700,3 +8701,123 @@ def make_interval(
             nanoseconds,
         )
     )
+
+
+def map(
+    dataframe,
+    func: Callable,
+    output_types: List[StructType],
+    output_column_names: Optional[List[str]] = None,
+    wrap_row=True,
+    **kwargs: Dict[str, Any],
+):
+    """Returns a new DataFrame with the result of applying `func` to each of the
+    rows of the specified DataFrame.
```

Comment:
can you also include that this function registers a UDTF internally and applies the given function to each column of the dataframe.

Reply:
Agreed, I just added this to the function documentation.